.github/actions/calculate-next-internal-version/next-prerelease.sh

```diff
@@ -1,24 +1,51 @@
 #!/usr/bin/env bash
-set -e
+set -euo pipefail
 
-if [ -n "$REPO_DIR" ]
-then
-  cd "$REPO_DIR"
+: "${NEXT_VERSION:?}"
+: "${PRERELEASE_TYPE:?}"
+: "${GITHUB_TOKEN:?Missing GITHUB_TOKEN}"
+
+# Use runner default if not explicitly set
+GITHUB_REPOSITORY="${GITHUB_REPOSITORY:-}"
+if [[ -z "$GITHUB_REPOSITORY" ]]; then
+  echo "GITHUB_REPOSITORY is not set"
+  exit 1
 fi
 
-echo "Fetching tags ... "
-git fetch --tags
+API_URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/tags"
+PER_PAGE=100
+PAGE=1
+MATCHING_TAG=""
+echo "::group::Fetching tags from GitHub API"
+
+while true; do
+  response=$(curl -s -H "Authorization: Bearer ${GITHUB_TOKEN}" \
```

Comment:
(nitpick info: the gh cli should be available on gh runners to help doing such calls in a more user-friendly way, I'm not asking you to change it, but please consider it for next time or if you think it'll improve maintainability of this code)